memory/src/main/scala/filodb.memory/data/OffheapLFSortedIDMap.scala

```diff
@@ -246,22 +286,173 @@ class OffheapLFSortedIDMapReader(memFactory: MemFactory, holderKlass: Class[_ <:
     innerBinSearch(0, mapLen)
   }
 
+  /**
+   * Acquire exclusive access to this map, spinning if necessary. Exclusive lock isn't re-entrant.
+   */
+  def acquireExclusive(inst: MapHolder): Unit = {
```

Comment:
Some metrics to track number of locks held?

Reply:
Only one exclusive can be held at a time. Do you mean a counter of all locks? Or do you mean gauge of shared locks held? It can cause contention.

Reply:
True. Wondering how we can provide some visibility to locking and contention via metrics effectively. Logging in and checking JMX is more of a debugging operation. If we can alert on this proactively, that would be good.

Reply:
I'm OK with a counter to count shared and exclusive lock operations.  Gauge is too invasive and slow probably.  However a counter on something which is concurrent can definitely cause contention also.   Adding a counter to `TimeSeriesStats` makes it shard-local and might not be too bad.

Reply:
Added some counters,